File: test-suite-kotlin/src/test/kotlin/io/micronaut/picocli/docs/cli/app/MyCliAppCommandTest.kt
package io.micronaut.picocli.docs.cli.app

import io.micronaut.configuration.picocli.PicocliRunner
import io.micronaut.context.ApplicationContext
import io.micronaut.context.env.Environment
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.io.ByteArrayOutputStream
import java.io.PrintStream

class MyCliAppCommandTest {

    @Test
    fun testWithCommandLineOption() {
        val baos = ByteArrayOutputStream()
        System.setOut(PrintStream(baos))
        ApplicationContext.run(Environment.CLI, Environment.TEST).use { ctx ->
            val args = arrayOf("-v")
            PicocliRunner.run(MyCliAppCommand::class.java, ctx, *args)

            Assertions.assertTrue(baos.toString().contains("Hi!"))
        }
    }
}

File: test-suite-python/src/test/python/micronaut/picocli/docs/cli/app/MyCliAppCommandTest.py
import logging
from io import StringIO
from typing import Annotated

from jakarta.inject import Inject
from micronaut.configuration.picocli import PicocliRunner
from micronaut.context import ApplicationContext
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from .MyCliAppCommand import MyCliAppCommand


@MicronautTest(environments=["cli"])
class MyCliAppCommandTest:

    ctx: Annotated[ApplicationContext, Inject]

    @Test
    def test_with_command_line_option(self):
        output = StringIO()
        handler = logging.StreamHandler(output)
        root = logging.getLogger()
        root.addHandler(handler)
        root.setLevel(logging.INFO)
        try:
            PicocliRunner.run(MyCliAppCommand, self.ctx, ["-v"])
        finally:
            root.removeHandler(handler)

        assert "Hi!" in output.getvalue()
